ml-agents/mlagents/trainers/tf_policy.py

```diff
@@ -364,82 +388,103 @@ def update_normalization(self, vector_obs: np.ndarray) -> None:
-        if done:
-            for k in value_estimates:
-                if self.reward_signals[k].use_terminal_states:
-                    value_estimates[k] = 0.0
-
-        return value_estimates
-
-    @property
-    def vis_obs_size(self):
-        return self.model.vis_obs_size
-
-    @property
-    def vec_obs_size(self):
-        return self.model.vec_obs_size
 
     @property
     def use_vis_obs(self):
-        return self.model.vis_obs_size > 0
+        return self.vis_obs_size > 0
 
     @property
     def use_vec_obs(self):
-        return self.model.vec_obs_size > 0
+        return self.vec_obs_size > 0
+
+    def _initialize_tensorflow_references(self):
+        self.value_heads: Dict[str, tf.Tensor] = {}
+        self.normalization_steps: Optional[tf.Variable] = None
+        self.running_mean: Optional[tf.Variable] = None
+        self.running_variance: Optional[tf.Variable] = None
+        self.update_normalization_op: Optional[tf.Operation] = None
+        self.value: Optional[tf.Tensor] = None
+        self.all_log_probs: tf.Tensor = None
+        self.log_probs: Optional[tf.Tensor] = None
+        self.entropy: Optional[tf.Tensor] = None
+        self.action_oh: tf.Tensor = None
+        self.output_pre: Optional[tf.Tensor] = None
+        self.output: Optional[tf.Tensor] = None
+        self.selected_actions: Optional[tf.Tensor] = None
+        self.action_holder: Optional[tf.Tensor] = None
+        self.action_masks: Optional[tf.Tensor] = None
+        self.prev_action: Optional[tf.Tensor] = None
+        self.memory_in: Optional[tf.Tensor] = None
+        self.memory_out: Optional[tf.Tensor] = None
+
+    def create_input_placeholders(self):
+        with self.graph.as_default():
+            self.global_step, self.increment_step_op, self.steps_to_increment = (
+                LearningModel.create_global_steps()
+            )
+            self.visual_in = LearningModel.create_visual_input_placeholders(
+                self.brain.camera_resolutions
+            )
+            self.vector_in = LearningModel.create_vector_input(self.vec_obs_size)
+            if self.normalize:
+                normalization_tensors = LearningModel.create_normalizer(self.vector_in)
+                self.update_normalization_op = normalization_tensors[0]
+                self.normalization_steps = normalization_tensors[1]
+                self.running_mean = normalization_tensors[2]
+                self.running_variance = normalization_tensors[3]
+                self.processed_vector_in = LearningModel.normalize_vector_obs(
+                    self.vector_in,
+                    self.running_mean,
+                    self.running_variance,
+                    self.normalization_steps,
+                )
+            else:
+                self.processed_vector_in = self.vector_in
+                self.update_normalization_op = None
+
+            self.batch_size_ph = tf.placeholder(
```

Comment:
Change `ph` to `placeholder`? Or is this a common tf abbreviation?

Reply:
It's pretty common in TF code to use `ph` - I can still change it if it's confusing though.